packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/mcp/intelligence_handler.py

```
@@ -948,3 +951,93 @@ def _cleanup_sessions_locked(self):
+        self,
+        cypher: str,
+        params: dict | None = None,
+    ):
+        """
+        Raw Cypher query (power users)
+        """
+
+        # basic safety check
+        forbidden = ["DELETE", "DETACH", "DROP"]
+        if any(word in cypher.upper() for word in forbidden):
+            raise ValueError("Dangerous query detected")
```

Ruff (0.15.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)